borsh-derive/src/internals/attributes/item/mod.rs

```diff
@@ -85,29 +83,49 @@ pub(crate) fn contains_use_discriminant(input: &ItemEnum) -> Result<bool, syn::E
     Ok(use_discriminant.unwrap_or(false))
 }
 
-pub(crate) fn contains_initialize_with(attrs: &[Attribute]) -> Option<Path> {
+pub(crate) fn contains_initialize_with(attrs: &[Attribute]) -> Result<Option<Path>, Error> {
     let mut res = None;
     let attr = attrs.iter().find(|attr| attr.path() == BORSH);
     if let Some(attr) = attr {
-        let _ = attr.parse_nested_meta(|meta| {
+        attr.parse_nested_meta(|meta| {
             if meta.path == INIT {
                 let value_expr: Path = meta.value()?.parse()?;
                 res = Some(value_expr);
-            } else if meta.path == USE_DISCRIMINANT {
+            }
+            if meta.path == USE_DISCRIMINANT || meta.path == CRATE {
                 let _value_expr: Expr = meta.value()?.parse()?;
             };
+
             Ok(())
-        });
+        })?;
     }
 
-    res
+    Ok(res)
+}
+
+pub(crate) fn get_crate(attrs: &[Attribute]) -> Result<Option<Path>, Error> {
+    let mut res = None;
+    let attr = attrs.iter().find(|attr| attr.path() == BORSH);
+    if let Some(attr) = attr {
+        attr.parse_nested_meta(|meta| {
+            if meta.path == CRATE {
+                let value_expr: Path = parsing::parse_lit_into(BORSH, CRATE, &meta)?;
+                res = Some(value_expr);
+            }
+            if meta.path == USE_DISCRIMINANT || meta.path == INIT {
+                let _value_expr: Expr = meta.value()?.parse()?;
+            };
```

Comment:
Can you help me understand why we need to parse something completely unrelated here?

Reply:
If we comment out these 115-117 highlighted lines, we get a failure in [test](https://github.com/near/borsh-rs/blob/crate_attribute/borsh-derive/src/internals/attributes/item/mod.rs#L389-L416) with following message:

```bash
---- internals::attributes::item::tests::test_init_function_with_use_discriminant_with_crate stdout ----
thread 'internals::attributes::item::tests::test_init_function_with_use_discriminant_with_crate' panicked at 'called `Result::unwrap()` on an `Err` value: Error("expected `,`")', borsh-derive/src/internals/attributes/item/mod.rs:412:20
```

The recommended parsing [helper](https://docs.rs/syn/latest/syn/struct.Attribute.html#method.parse_nested_meta)
parses the args `init = initialization_method, crate = "reexporter::borsh", use_discriminant=true` of `#[borsh(init = initialization_method, crate = "reexporter::borsh", use_discriminant=true)]` with following
[code](https://docs.rs/syn/latest/src/syn/meta.rs.html#384-399) somewhere down the way:

```rust
pub(crate) fn parse_nested_meta(
    input: [ParseStream](https://docs.rs/syn/latest/src/syn/parse.rs.html#223),
    mut logic: impl FnMut([ParseNestedMeta](https://docs.rs/syn/latest/src/syn/meta.rs.html#163-166)) -> Result<()>,
) -> Result<()> {
    loop {
        let path = input.call(parse_meta_path)?;
        logic(ParseNestedMeta { path, input })?;
        if input.is_empty() {
            return Ok(());
        }
        input.parse::<Token![,]>()?;
        if input.is_empty() {
            return Ok(());
        }
    }
}
```
It expects to completely consume input `ParseStream<'a>`, without skipping anything.
If we comment out ` let _value_expr: Expr = meta.value()?.parse()?;` it returns an error `input.parse::<Token![,]>()?;` on a new  loop iteration on a `=` punctuation, presumably originating [here](https://docs.rs/syn/latest/src/syn/token.rs.html#1006).